### PR TITLE
Restrict mini-14 to firing Ruger magazines

### LINKED ISSUE #15758: Ruger 223 mags should only be compatible with the Mini-14
As of 0.C-4543 (unmodded), the Ruger 223 magazine is accepted in STANAG weapons and the description specifically states it is STANAG compatible. I know it's pretty nit-picky, but IRL Ruger Mini-14 mags are proprietary and simply wouldn't work in any other weapon without a lot of modification. The main difference is how they lock in to place within the magazine well - a STANAG mag is pushed straight in whereas the Mini-14 mag is rocked into place (very similar to AK pattern rifles in that regard).

As such, I propose the following changes be made:
- Remove STANAG compatibility from Ruger 223 magazines
- Rename the magazine for accuracy, perhaps "Ruger Mini-14 magazine" (or "Ruger Mini-14 10 rd magazine" if higher capacity Mini-14 magazines are added down the line)
- Change the magazine description to reflect these changes

After searching here and the on the forums, I couldn't find any discussion as to why the magazine is STANAG compatible so I assumed this is an oversight - many apologies if it's intended for some reason!

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
```SUMMARY: Balance "Restrict Ruger Mini-14 to using Ruger magazines"```

#### Purpose of change
Fixes #15758
Ruger Mini-14 is notorious in that it does not accept STANAG magazines, make it so.

#### Describe the solution
Remove the STANAG magazines from the mini-14 magazine list.
Add some new Ruger-specific mangazines and spawns for them.

#### Describe alternatives you've considered
Leave it inaccurate and unbalanced.

### CODE REVIEW

**Comment by kevingranade** [comment]
Definitely needs testing, but I didn't do anything here that should put savegames at risk, though I do want to see what happens if a m4a1 with a stanag10 loaded is migrated across this commit.

I SUSPECT it will work just fine and will disallow reloading once it has been loaded once. which IMO is the best to hope for in this situation, but if it crashes or does anything too nasty it needs a fixup.

File: data/json/itemgroups/magazines.json [hunk 1]
```diff
@@ -169,7 +168,9 @@
       ["m14mag", 40 ],
       ["m14smallmag", 40 ],
       ["ruger1022mag", 100 ],
-      ["stanag10", 100 ],
+      ["stanag5", 50 ],
+      ["stanag10", 30 ],
```
Comment (Reclusive-reptile) [inline data/json/itemgroups/magazines.json]: You have stanag30 on here twice. I think you meant to add ruger30.

File: data/json/itemgroups/magazines.json [hunk 2]
```diff
@@ -169,7 +168,9 @@
       ["m14mag", 40 ],
       ["m14smallmag", 40 ],
       ["ruger1022mag", 100 ],
-      ["stanag10", 100 ],
+      ["stanag5", 50 ],
```
Comment (Reclusive-reptile) [inline data/json/itemgroups/magazines.json]: Ruger5?

File: data/json/items/magazine/223.json [hunk 3]
```diff
@@ -45,10 +45,27 @@
     "flags": [ "MAG_COMPACT" ]
   },
   {
+    "id": "ruger5",
+    "type": "MAGAZINE",
+    "name": "Ruger .223 5-round magazine",
+    "description": "A compact 5-round magazine used with the Ruger mini-14 rifle.",
```
Comment (Night-Pryanik) [inline data/json/items/magazine/223.json]:
```suggestion
    "description": "A compact 5-round magazine used with the Ruger Mini-14 rifle.",
```

File: data/json/items/magazine/223.json [hunk 4]
```diff
@@ -45,10 +45,27 @@
...
+    "flags": [ "MAG_COMPACT" ]
+  },
+  {
+    "//": "Used to be a STANAG 10 round, but switched to just be Ruger compatible",
     "id": "stanag10",
     "type": "MAGAZINE",
-    "name": "Ruger .223 magazine",
-    "description": "A compact 10-round magazine commonly used with semi-automatic STANAG compatible rifles.",
+    "name": "Ruger .223 10-round magazine",
+    "description": "A compact 10-round magazine used with the Ruger mini-14 rifle.",
```
Comment (Night-Pryanik) [inline data/json/items/magazine/223.json]:
```suggestion
    "description": "A compact 10-round magazine used with the Ruger Mini-14 rifle.",
```

File: data/json/items/magazine/223.json [hunk 5]
```diff
@@ -60,6 +77,22 @@
     "reliability": 9,
     "flags": [ "MAG_COMPACT" ]
   },
+  {
+    "id": "ruger30",
+    "type": "MAGAZINE",
+    "name": "Ruger .223 high-capacity magazine",
+    "description": "A high capacity magazine for use with the Ruger mini-14 rifle.",
```
Comment (Night-Pryanik) [inline data/json/items/magazine/223.json]:
```suggestion
    "description": "A high capacity magazine for use with the Ruger Mini-14 rifle.",
```

**Comment by ZhilkinSerg** [comment]
Looks good - existing magazines loaded to gun in saved game stay in the gun, can be fired and unloaded when empty. but cannot be reattached to gun which do not support this magazine.

**Pull request merged**